crypto/merkle/result.go

```diff
@@ -0,0 +1,55 @@
+// nolint: dupl
+package merkle
+
+import (
+	"bytes"
+	"encoding/json"
+
+	"github.com/gogo/protobuf/jsonpb"
+)
+
+//---------------------------------------------------------------------------
+// override JSON marshalling so we emit defaults (ie. disable omitempty)
+// note we need Unmarshal functions too because protobuf had the bright idea
+// to marshal int64->string. cool. cool, cool, cool: https://developers.google.com/protocol-buffers/docs/proto3#json
```

Comment:
I might be misunderstanding something but AFAIK int64, fixed64 and uint64 should be encoded as strings?
https://github.com/tendermint/tendermint/issues/3898#issuecomment-521344376

Reply:
Yep, we should probably remove the "note we need ..."